src/ts/components/account/ConfirmPhrase.tsx

```diff
@@ -56,21 +56,32 @@ class ConfirmPhrase extends React.Component<IConfirmPhraseProps, IConfirmPhraseS
   }
 
   isPhraseConfirmed = (): boolean => {
-    return !!this.props.accountStatus && !!this.props.accountStatus.newPhrase
-        && this.props.accountStatus.newPhrase === this.state.inputPhrase
+    return !!this.props.wallet && !!this.props.wallet.newPhrase
+        && this.props.wallet.newPhrase === this.state.inputPhrase
   }
 
   createAccount = () => {
     this.props.setError(null)
-    const { accountStatus } = this.props
-    if (accountStatus.newPassword) {
-      unlockWallet(accountStatus.newPassword).then(kp => {
+    const { wallet } = this.props
+    // add a new wallet using the same pass
+    if (!wallet.locked && wallet.created) {
+      if (wallet.newPhrase) {
+        createAccount(wallet.newPhrase, wallet.newAccountName).then(keyringPair => {
+          console.log('Account created! ', keyringPair)
+          this.setState({ keyringPair: keyringPair })
+        }).catch(err => {
+          this.props.setError(err)
+        })
+      }
+    }
+    if (wallet.newPassword) {
+      unlockWallet(wallet.newPassword).then(kp => {
         console.log('wallet unlocked')
         // update redux store state
         this.props.setLocked(false)
         console.assert(kp.length === 0, 'Should be an empty array')
-        if (accountStatus.newPhrase) {
-          createAccount(accountStatus.newPhrase, accountStatus.newAccountName).then(keyringPair => {
+        if (wallet.newPhrase) {
+          createAccount(wallet.newPhrase, wallet.newAccountName).then(keyringPair => {
             console.log('Account created! ', keyringPair)
             this.props.setCreated(true)
             this.setState({ keyringPair: keyringPair })
```

Comment:
`this.setState({ keyringPair })` shortcut might be used

Reply:
yes. updated.